modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/artifact/DeployableArtifact.java

```diff
@@ -49,4 +57,15 @@
    * @return MuleContext created from the artifact configurations files.
    */
   MuleContext getMuleContext();
+
+  /**
+   * @return the directory where the artifact content is stored.
+   */
+  File getLocation();
+
+  /**
+   * @return a service to test connection over configuration components.
+   */
+  ConnectivityTestingService getConnectivityTestingService();
```

Comment:
I don't agree much with that. If we do that then we are treating the Artifact as we do today with MuleContext that we don't know what users may be using it for.